src/ModelParser/VisibilityAwarePropertyAccessGuesser.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Liip\MetadataParser\ModelParser;
+
+use Liip\MetadataParser\Exception\ParseException;
+use Liip\MetadataParser\Metadata\PropertyAccessor;
+use Liip\MetadataParser\ModelParser\RawMetadata\PropertyVariationMetadata;
+use Liip\MetadataParser\ModelParser\RawMetadata\RawClassMetadata;
+
+/**
+ * Run this after all properties have been added by other parsers. This parser will modify existing property metadata
+ * referring to private properties by adding accessors if there are none. Said accessors are guessed from common
+ * getter/setter naming schemes such as `get{$property}`, `set{$property}`
+ */
+class VisibilityAwarePropertyAccessGuesser implements ModelParserInterface
+{
+    public function parse(RawClassMetadata $classMetadata): void
+    {
+        try {
+            $reflClass = new \ReflectionClass($classMetadata->getClassName());
+        } catch (\ReflectionException $e) {
+            throw ParseException::classNotFound($classMetadata->getClassName(), $e);
+        }
+
+        $this->parseProperties($reflClass, $classMetadata);
+    }
+
+    public function parseProperties(\ReflectionClass $reflClass, RawClassMetadata $classMetadata): void
+    {
+        if ($reflParentClass = $reflClass->getParentClass()) {
+            $this->parseProperties($reflParentClass, $classMetadata);
+        }
+
+        foreach ($reflClass->getProperties() as $property) {
+            if ($property->isPublic() || !$classMetadata->hasPropertyVariation($property->getName())) {
+                continue;
+            }
+
+            $variation = $classMetadata->getPropertyVariation($property->getName());
+            $currentAccessor = $variation->getAccessor();
+
+            if (!($currentAccessor->getSetterMethod() && $currentAccessor->hasSetterMethod())) {
```

Comment:
i am wrapping up this merge request. should the check be

```suggestion
            if (!($currentAccessor->hasGetterMethod() && $currentAccessor->hasSetterMethod())) {
```